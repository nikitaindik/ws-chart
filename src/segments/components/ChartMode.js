import React from 'react';

import Button from '../../components/Button';

import style from './ChartMode.module.css';

const oneDayInMs = 1 * 24 * 60 * 60 * 1000;
const sixHoursInMs = 6 * 60 * 60 * 1000;
const oneHourInMs = 1 * 60 * 60 * 1000;
const fiveMinsInMs = 5 * 60 * 1000;

const modes = [
  { label: 'Last 30 days', barSize: oneDayInMs },
  { label: 'Last 7 days', barSize: sixHoursInMs },
  { label: 'Last 24 hours', barSize: oneHourInMs },
  { label: 'Last 60 minutes', barSize: fiveMinsInMs },
];

const ChartMode = ({ activeBarSize, onChangeModeClick }) => (
  <div className={style.wrap}>
    {modes.map(mode => (
      <Button
        isActive={activeBarSize === mode.barSize}
        onClick={() => onChangeModeClick(mode.barSize)}
        key={mode.barSize}
      >
        {mode.label}
      </Button>
    ))}
  </div>
);

export default ChartMode;
