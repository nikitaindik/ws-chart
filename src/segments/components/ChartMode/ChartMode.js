import React from 'react';

import Button from '../../../components/Button';

import { time } from '../../constants';

import style from './ChartMode.module.css';

const modes = [
  { label: 'Last 30 days', barSize: time.ONE_DAY_IN_MS },
  { label: 'Last 7 days', barSize: time.SIX_HOURS_IN_MS },
  { label: 'Last 24 hours', barSize: time.ONE_HOUR_IN_MS },
  { label: 'Last 60 minutes', barSize: time.FIVE_MINUTES_IN_MS },
];

const ChartMode = ({ activeBarSize, onChangeModeClick }) => (
  <div className={style.wrap}>
    {modes.map(mode => (
      <Button
        isActive={activeBarSize === mode.barSize}
        onClick={() => onChangeModeClick(mode.barSize)}
        key={mode.barSize}
      >
        {mode.label}
      </Button>
    ))}
  </div>
);

export default ChartMode;
