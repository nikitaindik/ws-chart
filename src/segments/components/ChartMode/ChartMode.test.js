import React from 'react';
import { render } from 'react-testing-library';

import ChartMode from './ChartMode';

test('ChartMode snapshot', () => {
  const mockedProps = {
    activeBarSize: '360000',
    onChangeModeClick: jest.fn(),
  };

  const { container } = render(<ChartMode {...mockedProps} />);
  expect(container).toMatchSnapshot();
});
