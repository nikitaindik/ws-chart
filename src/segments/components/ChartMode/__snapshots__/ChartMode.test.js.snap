// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartMode snapshot 1`] = `
<div>
  <div
    class="wrap"
  >
    <button
      class="button"
    >
      Last 30 days
    </button>
    <button
      class="button"
    >
      Last 7 days
    </button>
    <button
      class="button"
    >
      Last 24 hours
    </button>
    <button
      class="button"
    >
      Last 60 minutes
    </button>
  </div>
</div>
`;
